slice.go

```diff
@@ -6,6 +6,20 @@ func CopySlice[T any](orig []T) []T {
 	return c
 }
 
+// FilterSlice returns a new slice containing only elements for which filter returns true.
+func FilterSlice[T any](slice []T, filter func(T) bool) []T {
+	result := make([]T, 0, len(slice))
+	for _, item := range slice {
+		if filter(item) {
+			result = append(result, item)
+		}
+	}
+	if len(result) == 0 {
+		return nil
```

Comment:
the analogous function in JS/TS returns an empty array/slice in this case. Not sure if we want that same behavior here or not? Was there a specific reason for going with `nil` vs `[]T{}`?

Reply:
In almost all cases in Go, nil works just fine.